django/db/models/expressions.py

```diff
@@ -547,6 +550,53 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None,
         return ResolvedOuterRef(self.name)
 
 
+class SliceableF(F):
+    """
+    An object that can add slicing notation to F expression.
+
+    Object resolves the column on which the slicing is applied, and then
+    applies the slicing if possible.
+    """
+    def __init__(self, f, k):
```

Comment:
please use longer/more descriptive names than `f` and `k`, like `f_obj` and `slice`